src/SMTLIB/Backends.hs

```diff
@@ -58,15 +58,15 @@ data Solver = Solver
     -- | An optional queue to write commands that are to be sent to the solver lazily.
     queue :: Maybe Queue,
     -- | The function used for logging the solver's activity.
-    log :: Builder -> IO ()
+    log :: LBS.ByteString -> IO ()
   }
 
 -- | Send a command in bytestring builder format to the solver.
 sendSolver :: Solver -> Builder -> IO LBS.ByteString
 sendSolver solver cmd = do
-  log solver $ "[send] " <> cmd
+  log solver $ toLazyByteString $ "[send] " <> cmd
   resp <- send (backend solver) cmd
-  log solver $ "[recv] " <> lazyByteString resp
+  log solver $ "[recv] " <> resp
```

Comment:
I think the logging function affords very little in this case.

A user could easily do
```Haskell
logQueryHoweverIWant cmd
resp <- sendSolver solver cmd
logResponseHoweverIWant resp
```

What problem does the logging API solve otherwise?

Reply:
@qaristote, what do you think of removing the logger function when initializing solvers?

Reply:
We can continue the discussion in https://github.com/tweag/smtlib-backends/pull/14

Reply:
It's true that it originally made sense because it was part of the larger `simple-smt` library, and it's not particularly useful here, so I'd be in favor of removing it.

One thing that could be a problem though would be sending commands in lazy mode: logging as you proposed could produce something like
```
[send] (set-option :produce-models true)
[send] (set-option :print-sucess true)) # intentional typo
[send] (assert (= 0 0))
[send] (check-sat)
[recv] (error "line 2 column 34: invalid command, '(' expected")
```
instead of
```
[send] (set-option :produce-models true)
(set-option :print-sucess true)) # intentional typo
(assert (= 0 0))
(check-sat)
[recv] (error "line 2 column 34: invalid command, '(' expected")
```